docs/guide/sending-email.md

```diff
@@ -0,0 +1,226 @@
+    pluralLabel: 'articles'
+  },
+  ...,
+  handlers(self) {
+    return {
+      'afterSave': {
+        async sendEmail(req, data) {
+          const options = {
+            from: 'admin@mysite.com',
+            to: 'editors@mysite.com',
+            subject: 'New Article added'
+          };
+          const template = 'email.html';
+          try {
+            await self.email(req, template, data, options);
+          return null;
+          } catch (err) {
+            self.apos.util.error('email notification error: ', err);
+          }
+        }
+      }
+    }
+  }
+};
+```
+
+<template v-slot:caption>
+/modules/article/index.js
+</template>
+
+</AposCodeBlock>
+
+
+<AposCodeBlock>
+
+```django
+<h1>A new article has been added to the site</h1>
+<p>Here is the blurb</p>
+{% data.title %}
+{% area data, 'blurb'}
+```
+
+<template v-slot:caption>
+/modules/article/views/email.html
+</template>
+
+</AposCodeBlock>
+
+In this example, we are creating a custom piece type that implements an article. The `handlers()` function is being passed the `afterSave` server event. This event is emitted any time this custom module saves a new article. We are using this signal to send out an email to the site editors. The data coming from the page is passed into the `email.html` template through the `data` argument to add the title and blurb for the editors to review.
+
+::: note
+Note that in the Nunjucks template, the area is passed in through the `data` context. It isn't being passed in through the `data.piece` or `data.page` context as it would for a normal Nunjucks template view.
+:::
+
+## Triggering email from a route
+
+In addition to using `handlers()` to trigger email delivery, you can use `apiRoutes()`. This can be used to send an email when a user gets redirected to a selected page or sends a POST request, like a form submission. All of the same arguments passed when invoking the `email()` method from a module need to be supplied when using the method in a route. In this case, we need to create options and our data object from our `req.body`.
+
+### Example usage
+
+<AposCodeBlock>
+
+```js
+module.exports = {
+  extend: '@apostrophecms/widget-type',
+  options: {
+    lable: 'contact'
+  },
+  ...,
+  apiRoutes(self) {
+    return {
+      post: {
+        submit: async function (req) {
+          if (!req.body.email || !req.body.name) {
+            throw self.apos.error('invalid');
+          }
+          const email = self.apos.launder(req.body.email);
+          const options = {
+            from: 'admin@mysite.com',
+            to: email,
+            subject: 'Thanks for your submission!'
+          };
+          const data = {
+            name: self.apos.launder(req.body.name)
+          };
+          const template = 'email.html';
+          try {
+            await self.email(req, template, data, options);
+            return null;
+          } catch (err) {
+            self.apos.util.error('email notification error: ', err);
```

Comment:
In this case, sending the email is the entire point, so don't catch; let the exception be caught automatically, logged, and reported to the caller as a 500 error.